public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -0,0 +1,253 @@
+import React, { useState } from 'react';
+import { sortBy } from 'lodash';
+import { GrafanaTheme, GrafanaTheme2, intervalToAbbreviatedDurationString, PanelProps } from '@grafana/data';
+import { CustomScrollbar, useStyles, useStyles2 } from '@grafana/ui';
+import { css, cx } from '@emotion/css';
+import { useAsync } from 'react-use';
+import alertDef from 'app/features/alerting/state/alertDef';
+import { AlertListOptions, ShowOption, SortOrder } from './types';
+
+import { fetchAllRules } from 'app/features/alerting/unified/api/prometheus';
+import { flattenRules, FlatRule, alertStateToState } from 'app/features/alerting/unified/utils/rules';
+
+export function UnifiedAlertList(props: PanelProps<AlertListOptions>) {
+  const [noAlertsMessage, setNoAlertsMessage] = useState('');
+
+  const currentAlertState = useAsync(async () => {
+    if (props.options.showOptions !== ShowOption.Current) {
+      return;
+    }
+
+    const alertsFromProm = await fetchAllRules();
```

Comment:
```suggestion
    const promRules = await fetchAllRules();
```

these are *rules* not *alerts*. A rule may or may not contain a list of alerts spawned from this rule. It's important to use proper naming or things will get confusing fast :)